dotnet/src/SemanticKernel/AI/Extensions/HttpClientExtensions.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Mime;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.AI.Extensions;
+
+/// <summary>
+/// HTTP Client extensions.
+/// </summary>
+public static class HttpClientExtensions
+{
+    /// <summary>
+    /// Sends HTTP request and handles response.
+    /// </summary>
+    /// <param name="httpClient">Instance of <see cref="HttpClient"/> to send request.</param>
+    /// <param name="requestUri">URI to send request.</param>
+    /// <param name="requestBody">Request body in <see cref="string"/> format.</param>
+    /// <returns>Returns deserialized response.</returns>
+    public static async Task<T> ExecutePostRequestAsync<T>(this HttpClient httpClient, Uri requestUri, string requestBody)
+    {
+        string responseJson;
+
+        try
+        {
+            using HttpContent content = new StringContent(requestBody, Encoding.UTF8, MediaTypeNames.Application.Json);
+
+            HttpResponseMessage response = await httpClient.PostAsync(requestUri, content).ConfigureAwait(false);
+
+            if (response == null)
+            {
+                throw new AIException(AIException.ErrorCodes.NoResponse, "Empty response");
```

Comment:
we can't use "AIExceptions" in a generic "HTTP" extension, or it will lead to incorrect error handling. If this code now is reusable outside the AI namespace, e.g. talking to any web api, the AIException will be inconsistent with the context where the HTTP request is running. 

It's definitely an good idea we'll have to consider the bigger picture before making this change, and since this is out of scope I would just cut it from the PR, and consider the improvement separately.